service/settings.py

```diff
@@ -198,13 +209,33 @@ def autodetect(self):
 
     def getProxySettings(self):
 
-        if self.getMode() == 0:
+        if self.getMode() == self.PROXY_MODE_NONE:
             return None
-        if self.getMode() == 1:
+        if self.getMode() == self.PROXY_MODE_AUTODETECT:
             return self.autodetect()
-        if self.getMode() == 2:
+        if self.getMode() == self.PROXY_MODE_MANUAL:
             return (self.getAddress(), int(self.getPort()))
 
+    def getProxyAuthDetails(self):
+        if self.getMode() == self.PROXY_MODE_NONE:
+            return None
+        if (self.serviceNetworkSettings["login"] is None) or (self.serviceNetworkSettings["password"] is None):
+            return None
+        # in all other cases, return tuple of (login, password)
+        return (self.serviceNetworkSettings["login"], self.serviceNetworkSettings["password"])
```

Comment:
Why are we returning a tuple here?  My first thought was to return a dict, but then looking at it again, why wouldn't we just:
`return self.serviceNetworkSettings`

If we want to return multiple values separately for some reason, you're better off using a dict so that the returned values actually have something meaningful to reference them by, and you're not having to do `returnValue[1]` type stuff.

`return {'login': self.serviceNetworkSettings["login"], 'password': self.serviceNetworkSettings["password"]}`